apps/web/src/components/Explore/TwitterSection.tsx

```diff
@@ -0,0 +1,105 @@
+import { useCallback, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { TwitterSectionQueryFragment$key } from '~/generated/TwitterSectionQueryFragment.graphql';
+import TwitterIcon from '~/icons/Twittericon';
+
+import colors from '../core/colors';
+import InteractiveLink from '../core/InteractiveLink/InteractiveLink';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { TitleDiatypeL } from '../core/Text/Text';
+import TwitterFollowingModal from '../Twitter/TwitterFollowingModal';
+import ExploreList from './ExploreList';
+
+type Props = {
+  title: string;
+  subTitle: string;
+  queryRef: TwitterSectionQueryFragment$key;
+};
+
+export default function TwitterSection({ queryRef, title, subTitle }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment TwitterSectionQueryFragment on Query {
+        socialConnections(
+          before: $twitterListBefore
+          last: $twitterListLast
+          socialAccountType: Twitter
+        ) @required(action: THROW) {
+          edges {
+            node {
+              __typename
+              ... on SocialConnection {
+                __typename
+                galleryUser {
+                  ... on GalleryUser {
+                    __typename
+                    ...ExploreListFragment
+                  }
+                }
+              }
+            }
+          }
+        }
+
+        ...ExploreListQueryFragment
+
+        ...TwitterFollowingModalFragment
+        ...TwitterFollowingModalQueryFragment
+      }
+    `,
+    queryRef
+  );
+
+  const nonNullUsers = useMemo(() => {
+    const users = [];
+
+    for (const edge of query.socialConnections?.edges ?? []) {
+      if (edge?.node?.__typename !== 'SocialConnection') return;
```

Comment:
I don't think you mean to return here. This is probably messing with your types.

Reply:
Updated the types!